File: 562backpackIV/backpackIV.cpp
class Solution {
public:
    /**
     * @param nums: an integer array and all positive numbers, no duplicates
     * @param target: An integer
     * @return: An integer
     */
    int backPackIV(vector<int> &nums, int target) {
        // write your code here
        std::vector<int> dp(target + 1);
        dp.front() = 1;
        for (auto const item: nums)
            for (std::size_t size{item}; size != target + 1; ++size) dp.at(size) += dp.at(size - item);
        return dp.back();
    }
};

File: 562backpackIV/backpackIV.py
import itertools

class Solution:
    """
    @param nums: an integer array and all positive numbers, no duplicates
    @param target: An integer
    @return: An integer
    """
    def backPackIV(self, nums, target):
        # write your code here
        dp = [1, *itertools.repeat(0, target)]
        for item in nums:
            for size in range(1, target + 1): dp[size] += dp[size - item] if size >= item else 0
        return dp[-1]
